eslint/babel-eslint-parser/test/typescript-estree.test.js

```diff
@@ -0,0 +1,190 @@
+          },
+        }).ast;
+
+        deeplyRemoveProperties(babelAST, PROPS_TO_REMOVE);
+        expect(babelAST).toEqual(tsEstreeAST);
+      }
+    }
+
+    beforeAll(() => {
+      // Use the version of TS-Espree that is a dependency of
+      // the version of typescript-eslint we are testing against.
+      const tsEstreePath = require.resolve(
+        "@typescript-eslint/typescript-estree",
+        {
+          paths: [path.dirname(require.resolve("typescript-eslint"))],
+        },
+      );
+
+      tsEstree = require(tsEstreePath);
+    });
+
+    it.each([
+      ["empty", ""],
+      ["boolean", "true"],
+      ["boolean", "false"],
+      ["numeric", "0"],
+      // ["bigint", "0n"],
+      ["string", `"string"`],
+      ["regexp without flag", `/foo/;`],
+      ["regexp u flag", `/foo/dgimsuy;`],
+      ["regexp v flag", `/foo/dgimsvy;`],
+      ["identifier", "i"],
+
+      ["logical NOT", `!0`],
+      ["bitwise NOT", `~0`],
+
+      ["function declaration", "function f(p) {}"],
+      ["function expression", "0, function f(p) {}"],
+      ["arrow function expression", "() => {}"],
+      ["async arrow function expression", "async () => {}"],
+
+      ["object method", "({ m() {} })"],
+      ["object property", "({ p: 0 })"],
+
+      ["class declaration", "class C {}"],
+      ["class expression", "0, class C {}"],
+      ["class method", "class C { m() {} }"],
+      ["class property", "class C { p; }"],
+      ["class static block", "class C { static {}; }"],
+
+      ["variable declaration", "var a = 0"],
+      [
+        "variable declaration destructuring",
+        "var [{ a: x = 0 }, ...b] = [{ a: 0 }]",
+      ],
+      // pending cloneIdentifier support
+      // ["variable declaration destructuring shorthand", "let { a = 0 } = { a: 0 }"]
+
+      ["assignment expression", "x = 1"],
+      // pending toAssignable support
+      // ["assignment expression destructuring", "[{ a: x = 0 }, ...b] = [{ a: 0 }]"],
+
+      ["async call expression", "async ([ x ])"],
+
+      ["import declaration", `import "foo"`],
+      ["import declaration default", `import foo from "foo"`],
+      // pending cloneIdentifier support
+      // ["import declaration named", `import { foo } from "foo"`]
+      ["import declaration named as", `import { foo as bar } from "foo"`],
+      [
+        "import declaration with attributes",
+        `import foo from "./foo.json" with { type: "json" }`,
+      ],
+
+      // pending cloneIdentifier support
+      // ["export declaration", `const foo = 0;export { foo }`],
+      ["export function declaration", `export function foo() {}`],
+      ["export function declaration", `export class foo {}`],
+
+      ["member expression", `foo.bar`],
+      ["call expression", `foo(bar)`],
+      ["new expression", `new foo`],
+    ])("%s: %s", (_, input) => {
+      parseAndAssertSame(input);
+    });
+
+    if (IS_BABEL_8()) {
+      it.each([
+        // ["class private method", "class C { #m() {} }"],
```

Comment:
This test is commented out due to https://github.com/typescript-eslint/typescript-eslint/issues/11022.